vNext/channels/applicationinsights-channel-js/src/EnvelopeCreator.ts

```diff
@@ -104,27 +103,49 @@ export abstract class EnvelopeCreator {
                 env.tags[CtxTagKeys.sessionId] = item.ext.app.sesId;
             }
         }
-        
-        // session.isFirst is not supported in CS 4.0
-        // if (item.tags[CtxTagKeys.sessionIsFirst]) { 
-        //     env.tags[CtxTagKeys.sessionIsFirst] = item.tags[CtxTagKeys.sessionIsFirst];
-        // }
 
         if (item.ext.device) {
             if (item.ext.device.id || item.ext.device.localId) {
                 env.tags[CtxTagKeys.deviceId] = item.ext.device.id || item.ext.device.localId;
             }
-        }
 
-        if (item.ext.ingest) {
-            if (item.ext.ingest.clientIp) {
-                env.tags[CtxTagKeys.deviceIp] = item.ext.ingest.clientIp; // mapping ingest to deviceIp
+            if (item.ext.device.ip) {
+                env.tags[CtxTagKeys.deviceIp] = item.ext.device.ip;
             }
         }
 
+
         if (item.ext.web) {
-            if (item.ext.web.browserLang) {
-                env.tags[CtxTagKeys.deviceLanguage] = item.ext.web.browserLang; // mapping browser language to device language
+            let web: IWeb = <IWeb>item.ext.web;
+
+            if (web.browserLang) {
+                env.tags[CtxTagKeys.deviceLanguage] = web.browserLang;
+            }
+            if (web.browserVer) {
+                env.tags[CtxTagKeys.deviceBrowserVersion] = web.browserVer;
+            }
+
+            if (web.browser) {
+                env.tags[CtxTagKeys.deviceBrowser] = web.browser;
+            }
+            env.data = env.data || {};
+            env.data.baseData = env.data.baseData || {};
+            env.data.baseData.properties = env.data.baseData.properties || {};
+
+            if (web.domain) {
+                env.data.baseData.properties['domain'] =web.domain;
+            }
+
+            if (web.isManual) {
+                env.data.baseData.properties['isManual'] = web.isManual.toString();
+            }
+
+            if (web.screenRes) {
```

Comment:
screenRes used to be in tags, is this change intentional?

Reply:
https://1dsdocs.azurewebsites.net/articles/schema/common/schema-pageviews-and-usage.html screenRes is in web extension per documentation.